### DISCUSSION TITLE
ArgoCD applications getting refresh very slow & application controller pod CPU usage is too high

### DISCUSSION THREAD

**Discussion by naveen-vf** [argoproj/argo-cd#13964] [Q&A]
We are facing issue with ArgoCD application refresh is taking too long more 5 min, it is keep on rotating. also observed that application controller pod CPU usage is too high more 1200mCPU. it is blocking most of our development work.

Env Details
ArgoCD version: 2.5.1
Number of applications are around 70
AWS Cloud EKS cluster

Tried below already :
- Restarting the pods
- increase the replicas of application controller , repo-server, argocd-server
- Enabeld gzip compression
- there is no orphan applications 

No proper solution found on the previosuly raised issues. any help is very much appreciated.

**Comment by ashutosh16** [comment]
Are you running controller in HA mode? if not then try to increase the number of pods  - https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#argocd-application-controller

**Reply by astraw99** [reply]
Followed by this doc, solved the refresh slow issue.
Thanks.

@naveen-vf 
We should both update the replicas for application-controller, repo-server, argocd-server and their ENV value.

My config:
application-controller:
replicas = 3
ARGOCD_CONTROLLER_REPLICAS = 3

argocd-server:
replicas = 3
ARGOCD_API_SERVER_REPLICAS = 3

repo-server:
replicas = 3
(no replica env found)

**Reply by busyguys** [reply]
It works for me. Thx
